src_erl/NerlnetApp/src/Source/sourceStatem.erl

```diff
@@ -110,19 +111,19 @@ state_name(_EventType, _EventContent, State = #source_statem_state{}) ->
 
 
 %% This cast receive a list of samples to load to the records batchList
-idle(cast, {batchList,WorkersList,Epochs, CSVData}, State) ->
+idle(cast, {batchList,WorkersList,NumOfBatches, CSVData}, State) ->
   EtsRef = get(source_ets),
   StatsEtsRef = get(source_stats_ets),
   MyName = ets:lookup_element(EtsRef, my_name, ?DATA_IDX),
   BatchSize = ets:lookup_element(EtsRef, batch_size, ?DATA_IDX),
   {NerlTensorBatchesList, NerlTensorType, SampleSize} = parser:parseCSV(MyName,BatchSize,CSVData), % TODO this is slow and heavy policy! pre parse in ETS a possible solution
   ets:update_element(EtsRef, workers_list, [{?DATA_IDX, WorkersList}]),
-  ets:update_element(EtsRef, epochs, [{?DATA_IDX, Epochs}]),
+  ets:update_element(EtsRef, num_of_batches, [{?DATA_IDX, NumOfBatches}]),
   stats:increment_messages_received(StatsEtsRef),
   ?LOG_NOTICE("Source ~p, workers are: ~p", [MyName, WorkersList]),
   ?LOG_NOTICE("Source ~p, sample size: ~p", [MyName, SampleSize]),
   ets:update_element(EtsRef, sample_size, [{?DATA_IDX, SampleSize}]),
-  ?LOG_INFO("Source ~p updated transmission list, total avilable batches to send: ~p~n",[MyName, length(NerlTensorBatchesList)]),
+  ?LOG_INFO("Source ~p updated transmission list, total avilable batches to send: ~p~n",[MyName, NumOfBatches]),
```

Comment:
@GuyPerets106 
Take the minimum value of the two to be the number of batches
min(NumOfBatches, length(NerlTensorBatchesList))
and please add LOGERROR if NumOfBatches > length(NerlTensorBatchesList)